experimental/Schemes/WeilDivisor.jl

```diff
@@ -0,0 +1,352 @@
+export WeilDivisor
+export scheme_type, ideal_sheaf_type, coefficient_ring_type, coefficient_type
+export scheme, components, coefficient_dict, coefficient_ring
+export in_linear_system
+
+export LinearSystem 
+export divisor, find_subsystem
+
+@Markdown.doc """
+    WeilDivisor
+
+A Weil divisor on a covered scheme ``X``; 
+stored as a formal linear combination over 
+some ring ``R`` of ideal sheaves on ``X``.
```

Comment:
What kind of ideal sheaf? Prime ideal sheaf?

Reply:
I would expect a prime ideal sheaf. (Prime in the coordinate ring ... not geometrically prime)
 Anything else can get quite complicated fast ... and if it is not a prime ideal but composite I am not sure that we can do anything useful with it? So I would rather insist on `I` being prime.